File: algorithms-data-structures/go/src/algorithms/mixed/common-char-count/common_test.go
package commoncharcount

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestCommonCharCount(t *testing.T) {
	var tests = []struct {
		s1 string
		s2 string
		x  int
	}{
		{"aabcc", "adcaa", 3},
		{"abca", "xyzbac", 3},
		{"zzzz", "zzzzzzz", 4},
		{"abca", "xyzbac", 3},
		{"a", "b", 0},
		{"a", "aaa", 1},
	}

	for _, test := range tests {
		assert.Equal(t, test.x, commonCharacterCount(test.s1, test.s2))
	}
}

File: algorithms-data-structures/ts/src/algorithms/mixed/common-char-count/common.test.ts
import {commonCharacterCount} from "./common";
import {test, expect} from "bun:test";

test("commonCharacterCount", () => {
  let s1 = "aabcc";
  let s2 = "adcaa";
  let res = commonCharacterCount(s1, s2); // 3
  expect(res).toEqual(3);
});

test("commonCharacterCount", () => {
  let s1 = "zzzz";
  let s2 = "zzzzzzz";
  let res = commonCharacterCount(s1, s2); // 4
  expect(res).toEqual(4);
});
